modules/consensus/accept.go

```diff
@@ -306,9 +306,12 @@ func (cs *ConsensusSet) managedAcceptBlocks(blocks []types.Block) (blockchainExt
 	if !chainExtended {
 		return false, modules.ErrNonExtendingBlock
 	}
-	// Send any changes to subscribers.
-	for i := 0; i < len(changes); i++ {
-		cs.updateSubscribers(changes[i])
+	// Check if there are any subscribers
+	if len(cs.subscribers) > 0 {
```

Comment:
I feel like I would prefer this check inside of `updateSubscribers`. Something like the following at the beginning of the function. That way we don't need to add the check every time if we call `updateSubscribers` in multiple places.
```
if len(cs.subscribers) == 0 {
    return
}
```